src/main/java/bio/terra/service/filedata/azure/tables/TableServiceClientUtils.java

```diff
@@ -50,12 +55,25 @@ public static boolean tableExists(TableServiceClient tableServiceClient, String
   }
 
   public static List<TableEntity> batchRetrieveFiles(
-      TableServiceClient tableServiceClient, String tableName, List<String> fileIdArray) {
+      TableServiceClient tableServiceClient, List<String> fileIdArray) {
     String filter =
         fileIdArray.stream()
             // maybe wrap or cause in parenthesis
             .map(refId -> String.format("fileId eq '%s'", refId))
             .collect(Collectors.joining(" or "));
-    return filterTable(tableServiceClient, tableName, filter);
+    return filterTable(tableServiceClient, StorageTableUtils.getDatasetTableName(), filter);
+  }
+
+  public static int getTableEntryCount(
+      TableServiceClient tableServiceClient, String tableName, ListEntitiesOptions options) {
+    if (options == null) {
+      options = new ListEntitiesOptions();
+    }
+    if (tableHasEntries(tableServiceClient, tableName, options)) {
+      TableClient tableClient = tableServiceClient.getTableClient(tableName);
+      PagedIterable<TableEntity> entities = tableClient.listEntities(options, null, null);
+      return Iterables.size(entities);
```

Comment:
Unfortunately, this is a very expensive way to get the size of the entities, but its the only using the storage tables API. This is going to consume the entire contents of the returned entities just to count them. In order to reduce the network traffic, we may want to set the `select` option to only grab the row key and partition key so that we're not streaming the entire entity contents over the network just to count it.

Reply:
Good point - what I really need to know here is if there is a unique entry in the storage table given the query parameters. I just find working with the PagedIterable a bit tricky, so my new solution looks a bit gross. But, it should be more efficient.